app/controllers/api/v1/trips_controller.rb

```diff
@@ -1,7 +1,35 @@
+require 'pry'
 class Api::V1::TripsController < ApplicationController
-  protect_from_forgery unless: -> { request.format.json? }
+  before_action :authorize_user, only: [:create, :destroy, :update]
```

Comment:
Nice helper method! Devise should have a helper method `authenticate_user!` to accomplish this check that a user is signed in, this line would look like:
```
before_action :authenticate_user!, only: [:create, :destroy, :update]
```
and it would make the same check that you defined - if the current user is signed in
"authenticate" is used to ensure a visitor is signed in, whereas "authorize" is used to verify if they have permission to a resource